docs/painless/painless-guide/painless-intro.asciidoc

```diff
@@ -0,0 +1,173 @@
+[[painless-intro]]
+=== A Painless Introduction
+
+==== Welcome to Painless
+
+Welcome to the Painless scripting language!
+
+If you have reached this documentation, you are probably looking to come up
+with a solution to your use-case outside of what the standard Elasticsearch
+APIs may provide. If so, you're in the right place.
+
+Painless is a scripting language designed to extend the Elasticsearch APIs
+under many different environments known as
+<<painless-contexts, script contexts>> allowing for solutions to more
+advanced use-cases. Each script context provides support for a different
+environment such as ingest or scoring and determines what data is available and
+how to access that data.
+
+==== Design philosophy
+
+Now, before we dive into the basics of writing a Painless script, let's discuss
+the design philosophy we have in mind when choosing the syntax and available
+features for the language.
+
+* Safety: Our primary goal is to ensure the security of your cluster. To
+that end, Painless uses a fine-grained allowlist with a granularity down to the
+members of a class. Anything that is not part of the allowlist simply does not
+execute. See the {painless}/painless-api-reference.html[Painless API Reference]
+for a complete list of available classes, methods, and fields per script
+context.
+* Performance: Our secondary goal is to ensure the best performance possible.
+Painless compiles directly into JVM bytecode to take advantage of all possible
+optimizations the JVM provides which are numerous. We also generally avoid
+features that require additional slower checks at runtime.
+* Syntax: Our final goal is to implement a syntax with a natural familiarity to
+anyone with some basic coding experience. Painless uses a subset of Java syntax
+with some additional sugar to improve readability and remove boilerplate. Use
+of a Java-like syntax has the additional benefit of a natural compilation path
+into JVM bytecode that the syntax and semantics of other languages tend to
+heavily complicate.
+
+With that said, let's jump into a few examples of queries using a Painless
+script.
+
+==== Hello, world!
+
+For our examples in this introduction we use the
+<<painless-execute-api, painless execute API>> which by default allows
+us to test a script with simple user-defined parameters and receive a result.
+
+===== Hello, world! as a simple REST query
+
+Let's take a look at a simple script and break out the individual pieces.
+
+[source,console]
+----
+POST /_scripts/painless/_execute <1>
+{
+  "script": { <2>
+    "source": "return 'Hello, world!';" <3>
+  }
+}
+----
```

Comment:
It would also be great if we could reduce this and other examples to just the script itself `return 'Hello, world!';` and surface a link for loading this in Painless Lab, the same way you can load it into Console. Feel free to open an enhancement request issue in the Kibana repo and mention me in it. In the meantime, maybe these docs could mention Painless Lab, to help people run these examples easily?